src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,87 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
 
-public class TournamentStats {
+class TournamentStats {
+    private static final String FILE_NAME = "stats.csv";
+    private final List<Player> playerList = new ArrayList<>();
 
     void run(Scanner scanner) {
-        // tutaj dodaj swoje rozwiązanie
-        // użyj przekazanego scannera do wczytywania wartości
+        try {
+            mainLoop(scanner);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.err.println("Podałeś błędne wartości spróbuj jeszcze raz ponownie uruchamiając program.");
+        }
+    }
+
+    private void mainLoop(Scanner scanner) {
+        boolean stop = false;
+        readDataFromUser(scanner, stop);
+        Comparator<Player> comparator = createComparatorWithOrder(scanner);
+        playerList.sort(comparator);
```

Comment:
Zmieniłbym jeszcze to, żeby lista nie była polem, tylko zmienną lokalną. W takiej sytuacji metoda readDataFromUser() powinna zwracać listę 
```
List<Player> playerList = readDataFromUser(scanner, stop);
```
W takiej sytuacji lepiej widać skąd się co bierze i nie trzeba skakać po kodzie w poszukiwaniu deklaracji